crates/project_panel/src/project_panel.rs

```diff
@@ -2242,8 +2281,10 @@ impl ProjectPanel {
             .selection
             .map_or(false, |selection| selection.entry_id == entry_id);
         let width = self.size(cx);
-        let filename_text_color = if details.is_diagnostic_error {
-            Color::Error
+        let filename_text_color = if settings.show_diagnostics != ShowDiagnostics::Off
+            && details.diagnostics_color.is_some()
+        {
+            details.diagnostics_color.unwrap()
         } else {
             entry_git_aware_label_color(details.git_status, details.is_ignored, is_marked)
         };
```

Comment:
Let me know if you know a nicer way to write this :) I don't really like the `unwrap()`